File: packages/react-native-nitro-image/android/src/main/java/com/margelo/nitro/image/HybridViewWithChildren.kt
package com.margelo.nitro.image;

import android.content.Context
import android.graphics.Color
import androidx.annotation.Keep;
import com.facebook.proguard.annotations.DoNotStrip;
import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.views.view.ReactViewGroup


/*
 * ViewWithChildrenImpl extends ReactViewGroup so that style properties like flex,
 * justifyContent, alignItems, padding etc.. can be applied.
 * Feel Free to use LinearLayout or any other ViewGroup that suits your needs.
 */
class ViewWithChildrenImpl: ReactViewGroup {
    constructor(context: Context) : super(context)
}

@Keep
@DoNotStrip
class HybridViewWithChildren(val context: ThemedReactContext): HybridViewWithChildrenSpec() {
    // View
    override val view = ViewWithChildrenImpl(context)

    // Props
    override var colorScheme: ColorScheme = ColorScheme.LIGHT
    override var someCallback: () -> Unit = {}

    // Methods
    override fun someMethod(): Unit {
        someCallback()
    }
}

File: packages/react-native-nitro-test/ios/HybridViewWithChildren.swift
//
//  HybridViewWithChildren.swift
//  NitroTest
//
//  Created by Patrick Kabwe on 06/05/2025.
//

import Foundation
import UIKit

class HybridViewWithChildren : HybridViewWithChildrenSpec {
  // UIView
  var view: UIView = UIView()
    
    override public init() {
        // view.backgroundColor = .red
    }

  // Props
  var colorScheme: ColorScheme = .light
  var someCallback: () -> Void = { }

  // Methods
  func someMethod() throws -> Void {
    someCallback()
  }
}
